checks/apps/namd/namd_check.py

```diff
@@ -1,133 +1,296 @@
+
+        tarsource = os.path.join(
+            self.namd_sources.stagedir,
+            f'NAMD_{self.namd_sources.version}_Source.tar.gz',
+        )
+
+        # Extract source code and compiler bundled Charm++
+        self.prebuild_cmds = [
+            f'tar --strip-components=1 -xzf {tarsource} -C {self.stagedir}',
+            'tar -xf charm-8.0.0.tar',  # INFO: Depends on the NAMD version
+            'cd charm-8.0.0',
+            f'./build charm++ multicore-linux-arm8 gcc --with-production --enable-tracing -j {self.build_system.max_concurrency}',
+            'cd ..',
+            # Link against tcl8.6 (provided by the UENV)
+            'sed -i \'s/-ltcl8.5/-ltcl8.6/g\' arch/Linux-ARM64.tcl',
+        ]
+
+        prefix = '/user-environment/env/develop-single-node'
+
+        self.build_system.config_opts = [
+            'Linux-ARM64-g++.cuda',
+            '--charm-arch multicore-linux-arm8-gcc',
+            f'--charm-base {self.stagedir}/charm-8.0.0',
+            '--with-tcl',
+            f'--tcl-prefix {prefix}',
+            '--with-fftw',
+            '--with-fftw3',
+            f'--fftw-prefix {prefix}',
+        ]
+
+        if self.uarch == 'gh200':
+            self.build_system.config_opts += [
+                '--with-single-node-cuda',
+                '--with-cuda',
+                '--cuda-gencode arch=compute_90,code=sm_90',
+            ]
+
+    @sanity_function
+    def validate_test(self):
+        self.namd_executable = os.path.join(
+            self.stagedir, 'Linux-ARM64-g++.cuda', 'namd3'
+        )
+        return os.path.isfile(self.namd_executable)
+
+
+class NamdCheck(rfm.RunOnlyRegressionTest):
+    descr = 'NAMD STMV Benchmark'
+    test_name = 'stmv'
+    valid_systems = ['*']
+    executable = 'namd3'
+    maintainers = ['SSA']
+    namd_input = fixture(namd_input_download, scope='session')
+
+    @run_before('run')
+    def prepare_run(self):
+        tarsource = os.path.join(
+            self.namd_input.stagedir,
+            f'NAMD-uenv.tar.gz',
+        )
+
+        # Extract source code and compiler bundled Charm++
+        self.prerun_cmds = [
+            f'tar -xzf {tarsource} -C {self.stagedir}',
+        ]
+
+        self.uarch = uenv.uarch(self.current_partition)
+        config = slurm_config[self.test_name][self.uarch]
+
+        self.num_tasks_per_node = config['ntasks-per-node']
+        self.num_tasks = config['nodes'] * self.num_tasks_per_node
+        self.num_cpus_per_task = config['cpus-per-task']
+        self.ntask_per_core = 1
+        self.time_limit = config['walltime']
+
+        self.executable_opts = [
+            f'+p {self.num_cpus_per_task}',
+            '+pmepes 5',
+            '+setcpuaffinity',
+        ]
+
+        if self.uarch == 'gh200':
+            self.executable_opts += ['+devices 0,1,2,3']
+            # It is required to make all the GPUs visible
+            self.extra_resources = {
+                'gres': {'gres': 'gpu:4'},
```

Comment:
>  it will be simply ignored ?

https://reframe-hpc.readthedocs.io/en/stable/regression_test_api.html#reframe.core.pipeline.RegressionTest.extra_resources